pkg/network/ebpf/c/protocols/http2/decoding.h

```diff
@@ -450,15 +415,41 @@ static __always_inline bool read_frame(pktbuf_t pkt, http2_frame_t *current_fram
 
 // Fixes an incomplete frame header. The function is trying to read the remaining of a split frame header.
 static __always_inline bool fix_incomplete_frame_header(pktbuf_t pkt, incomplete_frame_t *incomplete_frame, http2_frame_t *current_frame) {
-    pktbuf_fix_header_frame(pkt, (char*)current_frame, incomplete_frame);
-    if (format_http2_frame_header(current_frame)) {
-        pktbuf_advance(pkt, incomplete_frame->header_bytes_left);
-        incomplete_frame->header_bytes_left = 0;
-        return true;
+    // Get the current frame header bytes left.
+    __u32 header_bytes_left = incomplete_frame->header_bytes_left;
+    // If we don't have enough bytes in the packet to read the remaining header, we read the remaining bytes.
+    if (pktbuf_data_offset(pkt) + header_bytes_left > pktbuf_data_end(pkt)) {
+        header_bytes_left = pktbuf_data_end(pkt) - pktbuf_data_offset(pkt);
     }
-    // We didn't read a valid frame header, so we're marking it for deletion.
-    incomplete_frame->header_bytes_left = 0;
-    return false;
+    // Copy what we already have in the incomplete frame buffer to the current frame.
+    bpf_memcpy(current_frame, incomplete_frame->buf, HTTP2_FRAME_HEADER_SIZE);
+    // Read the remaining bytes from the packet.
+    pktbuf_read_into_buffer_http2_frame_header(incomplete_frame->buf, pkt, pktbuf_data_offset(pkt));
+
+    // Converting the current frame to a char pointer to copy the remaining bytes.
+    char *ptr = (char*)current_frame;
+
+    #pragma unroll(HTTP2_FRAME_HEADER_SIZE)
```

Comment:
Wondering about the effect of this code on instruction count. Since we're adding this part with the `#pragma unroll,` maybe we should validate the impact of this change on the instruction count?